tests/unit/test_variants.py

```diff
@@ -70,3 +73,112 @@ def testPhasesetZero(self):
 
     def testGenotypeHaploid(self):
         self.verifyGenotypeConversion("1", "376", [1], None)
+
+
+class TestAbstractVariantSet(unittest.TestCase):
+    """
+    Unit tests for the abstract variant set.
+    """
+    def setUp(self):
+        self._vsname = "testVariantSet"
+        self._backend = backend.AbstractBackend()
+        self._dataset = datasets.AbstractDataset(self._backend)
+        self._variantSet = variants.AbstractVariantSet(
+            self._dataset, self._vsname)
+
```

Comment:
I needed an actual variant to pass through to some methods (see PR description) because `hashVariant(v)` and `toProtocolElement()` just die because certain values are `None` - i suppose if we only want to test the methods as is then validating method failure is a fine enough test. let me know and ill remove these parts (or move to simulatorTest) 
https://github.com/jsh2134/server/commit/341ad1c19ee490c608174c510ce2fba7f4ec8136#diff-7858b4c3a3f24acfcfa6d89486e1544cR161 and https://github.com/jsh2134/server/commit/341ad1c19ee490c608174c510ce2fba7f4ec8136#diff-7858b4c3a3f24acfcfa6d89486e1544cR170